Version 1.9: Fix #18, #33, #34, #36, #38, #40, #41, #44 and many other fixes. Add CHANGELOG.md

samcv opened the pull request:
* Add CONTRIBUTING.md
* Automate the generation of all of the `Q[…]`, `q[…]`, `qq[…]` constructs using a script.
* Automate the generation of all ```#`[ ]``` multi-line comments using the same script.
* As an awesome bonus this allows us to add new types of brackets by only adding those brackets to the script for both comments and `Q`, `q` and `qq`

# v1.9

### Regex
* Let regex using `m/ /` or `rx/ /` appear anywhere and span multiple lines.
* Fancy single quotes in regex now actually quote `/‘…’/`. Fixes issue #45
* Let bare regex `/ /` appear after `=>`. Mostly fixes issue #35

### Heredocs
* Fix fancy heredoc delimiters like `q:to/📝🔚/`; Fixes issue #36

## Quoting
* Get left single and right single quotation marks working with nesting `‘testing 1 ‘2’ 3’`
* Get left double and right double quotation marks working with nesting `“testing 1 “2” 3”`
* Get left single and right single quotation marks working inside regex. Fixes issue #48
* Have the `｢this｣` quoting construct (which is a shortcut for `Q[]` quoting) work.
* Get halfwidth (standard) width corner brackets working for these strings: `｢nesting｢works｣｣`.
https://github.com/samcv/lalannguage-perl6fe/blob/testing/CHANGELOG.md#q-qq-and-q-quoting-constructs
### `Q`, `qq` and `q` quoting constructs
* `Q`, `qq` and `q` constructs using `(( ))`, `{{ }}`, `<< >>`, `[ ]`, `[[ ]]`, `( )`,
`{ }`, `/ /`,  `< >` now properly escape delimiters and other escape sequences. Fixes issue #46
* Get `Q`, `qq` and `q` constructs using arbitrary non-word characters working. Example: q%…%
* Add `Q`, `qq` and `q` constructs using `‘…’` and  `“…”`. Example: q“…”
* Add `Q`, `qq` and `q` constructs using `｢…｣`, `(((…)))`, `{{{…}}}`, `[[[…]]]` and`「…」`.


### Variables
* Fix certain `$<variables>` inside interpolated strings not syntax highlighting. Fixes issue #41
* Make `:token` `:regex` `:rule` not explode the highlighter. Fixes issue #33
* ￼Get all the forms of qq[] properly highlighting variables (interpolation). Fixes issue #18

### Methods
* Highlight `.categorize`, `.prepend`, `.parse-base`, `categorize-list` and `.antipairs` methods. Fixes issue #38
* Add `IO` related methods.
* Make sure some more things highlight properly if used as keys for pairs.
* Add `long`, `longlong`, `Pointer` and `CArray` data types
* Make sure `/` As a metaoperator `[/]` doesn't break syntax highlighting Fixes #34

### Comments
* Get all types of ```#`(``` Multiline comments working when there is leading whitespace before the #\`( Fixes issue #44
* Get these multiline comments working:
  ```#`((( )))```, ```#`{{{ }}}```, ```#`[[[ ]]]```, ```#`<< >>```, ```#`< >```, ```#`“ ”```, ```#`‘ ’```

### Pod
* Pod comments now highlight properly working when there is leading whitespace.
* Make sure pod after `=para` and `=for` immediately stop that block. Make sure formatting doesn't run on further. Fixes issue #51
* Make sure pod after all other abbreviated forms like `=head` highlight as comments as well. Fixes issue #50

samcv committed:
Add qq_*string_content sections

samcv committed:
Get all the forms of qq[] properly highlighting variables (interpolation)

samcv committed:
Get “Nested “quotes” working” Left + Right Double Quotes

samcv committed:
Get left single and right single nested quotes working

samcv committed:
Get hw (standard) width corner brackets working ｢nesting｢works｣｣ too

samcv committed:
Fixes MadcapJake/language-perl6fe/#41 fixes certain variables inside double quotes and other interpolated strings

Such as "$<myvar>" for example

samcv committed:
Fix MadcapJake/language-perl6fe/#33 make :token :regex :rule not explode

samcv committed:
Fix MadcapJake/language-perl6fe/#38

Add antipairs, categorize and prepend to the methods that will
be highlighted.

MadcapJake commented:
Very nice! At work now, but maybe we can get 1.9.0 released tonight! Will be home in approx 10 hours to merge this.

samcv committed:
Get multiline comments #`((( #`{{{ and #`[[[ working

samcv committed:
Get multiline comments with whitespace working: #` ( #` { etc.

samcv committed:
Make multiline comments allow leading whitespace

Fix MadcapJake/language-perl6fe/#44
Example:
   \#`(  )

samcv commented:
@MadcapJake today I will work on making the list of things fixed in the PR text be something nice we can use for the official changelog.

samcv committed:
Get "$<variable" working.

Was already working for “ ” fancy quotes.

samcv committed:
Add categorize-list and parse-base as methods

samcv committed:
Add CArray, Pointer, long and longlong data types

samcv committed:
Fix fancy heredoc delimiters q:to/📝🔚/;

Fixes MadcapJake/language-perl6fe/#36

samcv committed:
Add a changelog (CHANGELOG.md)

samcv committed:
Fix typo in changelog

samcv committed:
Have '/' as a metaoperator [/] not break highlighting

Fixes MadcapJake/language-perl6fe/#34

samcv committed:
Update CHANGELOG.md

samcv committed:
Get q< \< \> > and qq<> working with escaping < or >

samcv committed:
Get q' \'', qq' \'' and Q' \'' working with escaping

samcv committed:
Get q" " and qq" " working with escaped delimiters

samcv committed:
Get q{ } and qq{ } working with escaped characters

samcv committed:
Fix Reg. in last few commits highlighting q<\t> and others

samcv committed:
Fix qq<< \<< \>> >> delimiter escaping and others

samcv committed:
Fix qq{{ }} and q<< >> escaping

samcv committed:
Escaping for qq<< >> and minor fix for q<< >>

samcv committed:
Get q(( )) and q[[ ]] escaping properly + a few other fixes

samcv committed:
Get qq(( )) escaping working

samcv committed:
Get q[[ ]] escaping working properly

samcv committed:
Various fixes to the escaping. All but q" " and q' ' are fixed

samcv committed:
Let regex using m/ / or rx/ / appear anywhere and across lines

Before only regex preceeding = , ( ) ~ symbols and whitespace
would highlight as a regex. Now you can put the regex anywhere
and it can span multiple lines if you put m or rx + any adverbs
in front of the term.
Does not fix MadcapJake/language-perl6fe/#35 but now you can use
rx or m to allow your regex to span multiple lines.

samcv committed:
Update changelog about m/ / and rx/ /

Being allowed to appear everywhere now.

samcv committed:
Allow bare regex / … / to work immediately after `=>`

Fixes MadcapJake/language-perl6fe/#35

samcv committed:
Clean up the grammar in the Changelog

samcv committed:
Fix up changelog to prepare for the next release.

samcv commented:
@MadcapJake v1.9 is ready to be release now 🌟 🎉

If I could also get the ability to add labels and close issues in the issue tracker that would be great. Thanks.

samcv committed:
Make sure q() qq() and Q() don't highlight. Only valid as q ( )

qq ( ) and Q ( ) with whitespace

samcv committed:
q '…' qq '…' and Q '…' need whitespace between them, so enforce this

samcv committed:
A few fixes. Even with (?x) spaces in char classes in Atom grammars

the spaces are significant. Remove them in the grammar file.

samcv committed:
Fix qq[[ ]] escaping

samcv committed:
Make sure traits don't highlight when used as keys in pairs

Fixes MadcapJake/language-perl6fe/#47
Example: `my %hash = binary => 2

samcv committed:
q"…" used to include interpolation, which was incorrect. Fix that.

samcv committed:
Fix ‘ ’ quotes inside of regex

rx/ ‘test’ /

samcv committed:
Update changelog

samcv committed:
Escapes aren't allowed in ｢…｣ quoting. Fix this.

｢\｣ used to not work properly. Since in Q[…] style quoting
you are not allowed to escape anything, even the delimiters.

samcv committed:
Make sure some more things highlight properly if used as keys.

These: fatal|internals|MONKEY\\-TYPING|nqp|
strict|trace|worries|invocant|parameters|experimental|
cur|soft|variables|attributes|v6(?:\\.\\w)*|lib|Test|NativeCall

samcv committed:
Make sure the 'types' don't highlight as types if used as keys

samcv committed:
Make sure die|fail|try|warn don't highlight when used as keys

samcv committed:
Make sure more keywords can be used as keys. Add trait `raw`

samcv committed:
Update changelog.

Fixed MadcapJake/language-perl6fe/#48

samcv committed:
Add a bunch of `IO` methods.

Also revert change that was incorrect. #` ( should only
comment for that line only not for further lines.

zoffixznet commented:
@MadcapJake when is this getting merged? Since GitHub is using this, we need fixes badly.

Any way you could give @samcv contributor perms so they could release?

samcv committed:
Update changelog

samcv committed:
Make Pod comments highlight properly when there is leading whitespace

samcv committed:
Add CONTRIBUTING.md with some helpful tips!

samcv committed:
Make pod after =para and =for immediately stop that block

* Use positive lookahead as well to make sure it doesn't capture the
next block or the empty line.
* Also add some comments
+ Fixes MadcapJake/language-perl6fe/#51

samcv committed:
Update Changelog

samcv committed:
Fix Pod comments after abbreviated forms like `=head`

Fixes MadcapJake/language-perl6fe/#50

samcv committed:
Start reforming the grammars for the q/qq quoting constructs

Also add a template file for creating these quoting constructs
This has the potential to allow us in the future to automate
generation of perl6fe.cson. This can allow us to add all of the
quoting constructs!

samcv committed:
All Q[] q[] qq[] constructs now properly escape and are autogenerated

Fixes MadcapJake/language-perl6fe/#46

samcv committed:
Add test file for q/qq/Q quoting constructs

samcv committed:
Add the replace script

samcv committed:
Get `Q`, `qq` and `q` constructs working with arbitrary non-word characters

Example: q%…%

samcv committed:
Add myself to the LICENSE for Ⓒ Copyright

samcv committed:
Clean up some of the links in the README.md

samcv committed:
Add Atom's swap files to .gitignore

samcv committed:
Add q, qq, Q‘…’ and q, qq, Q“…” quoting constructs.

samcv committed:
Update changelog

samcv committed:
Fix typo in changelog

samcv committed:
Automate generating #`( multi-line comments

Add #`<< >>, #`< >, #`“ ”, #`‘ ’
as a result

samcv committed:
Automate the rest of the multi-line comments. Magically…

We also get q, qq, Q quoting using these, with no
extra effort!
`｢…｣`, `(((…)))`, `{{{…}}}`, `[[[…]]]` and`「…」`

samcv committed:
Get nested fancy single quotes working in regex

samcv committed:
Fix edge case where q/qq/Q could start highlight erronerously

Calling a function called `use-for-q` by doing:
`use-for-q @delimiters` would start highlighting because
the @ will match the q_any rules. This is actually totally valid
and we still need to highlight `@` potentially. It's possible
this could also trigger for the other delimiters too, so apply
this to all of them, and only highlight if the previous character
is whitespace or we're at the start of the line.

samcv committed:
Revert that change in the README

samcv committed:
Fix typo \s should have been \\S for the Q quoting forms

samcv committed:
Remove `'` from non-word characters since they're allowed in var

variables names.

samcv committed:
Apply previous fix for edgecases to all the quoting constructs

samcv committed:
Actually revert readme

samcv committed:
Get array subscripts working in interpolated strings

Fixes half of MadcapJake/language-perl6fe/#24

samcv committed:
Prepare for v1.9

samcv committed:
Reference issue number in changelog

samcv committed:
Revert "Prepare for v1.9"

This reverts commit 73972719afdd8ac39ebf3a6f281fe2fb0d38066f.

samcv committed:
Move all the quoting constructs into their own file.

samcv committed:
Update the <word quote> section. Add prelim support for \x[…] …

Also make sure numbers at the start of a line highlight,
as well as highlight immediately after a `÷` division sign
a `×` multiplication sign or a `*`
Working toward MadcapJake/language-perl6fe/#39
For now \x[…] only highlights in interpolated strings.

samcv committed:
Add more tips to CONTRIBUTING.md

samcv committed:
Add method names. Make sure keys never highlight for pairs
